_posts/2022-12-07-jberet-quarkus.md

```diff
@@ -0,0 +1,123 @@
+
+This test uses the Quarkus test framework to set up the service running environment, and it will start the Quarkus before test, and then register the components and run the test. Here is the setup part of the test:
+
+```java
+@RegisterExtension
+static QuarkusUnitTest TEST = new QuarkusUnitTest()
+    .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+        .addClasses(DummyBatchlet.class)
+        .addAsManifestResource("batchlet.xml", "batch-jobs/batchlet.xml"));
+```
+
+In the above code, it creates the test JAR archive, and it puts the `batchlet.xml` into the JAR file. This `batchlet.xml` is defined in the test code directory:
+
+- https://github.com/quarkiverse/quarkus-jberet/blob/main/core/deployment/src/test/resources/batchlet.xml
+
+The `batchlet.xml` defines a basic `batchlet` for testing:
+
+```xml
+<job id="batchlet-job" xmlns="http://xmlns.jcp.org/xml/ns/javaee" version="1.0">
+    <step id="batchlet-step">
+        <batchlet ref="batchlet">
+            <properties>
+                <property name="name" value="#{jobParameters['name']}"/>
+            </properties>
+        </batchlet>
+    </step> 
+</job>
+```
+
+In the above job definition, it defines a `batchlet` which refers to an implementation just called `batchlet`, and it defines a property called `name`, which reads the value from a parameter `#{jobParameters['name']}`. The `batchlet` and its `name` are all defined in the `BatchletTest` code:
+
+```java
+@Named("batchlet")
+@Dependent
+public static class DummyBatchlet implements Batchlet {
+      @Inject
+      @BatchProperty(name = "name")
+      String name;
+
+      @Override
+      public String process() {
+          if (!name.equals("david")) {
+              throw new RuntimeException("Unexpected value injected to 'name': " + name);
+          }
+          return BatchStatus.COMPLETED.toString();
+      }
+
+      @Override
+      public void stop() {
+      }
+}
+```
+
+In the above `DummyBatchlet`, it defines the `process()` method that will be run in the job. And the value `name` of the property will be injected during the test running process. Here is one of the test methods inside the test class:
+
+```java
+@Test
+public void runBatchletJob() {
+    Properties jobParameters = new Properties();
+    jobParameters.setProperty("name", "david");
+    JobOperator jobOperator = BatchRuntime.getJobOperator();
+    long executionId = jobOperator.start("batchlet", jobParameters);
+
+    await().atMost(5, TimeUnit.SECONDS).until(() -> {
+        JobExecution jobExecution = jobOperator.getJobExecution(executionId);
+        return BatchStatus.COMPLETED.equals(jobExecution.getBatchStatus());
+    });
+}
+```
+
+The `runBatchletJob` setup the `name` property, and then start the `batchlet` job, and then it will wait for 5 seconds for the job to be finished, and check the job status properly. Now we can run the test by running the following commands in the test directory:
```

Comment:
make sure the job name `batchlet` is consistent with the one used in the above job xml file.

Reply:
Same as above, I modified the text to be more accurate.